File: ruby/raindrops/raindrops.rb
module Raindrops
  RAINDROPS = {
    3 => 'Pling',
    5 => 'Plang',
    7 => 'Plong',
  }
  def self.convert(number)
    (rainword = RAINDROPS.map {|divisor, sound| number % divisor == 0 ? sound : ""}.join)
      .empty? ? number.to_s : rainword
  end
end

File: javascript/raindrops/raindrops.js
//
// This is only a SKELETON file for the 'Raindrops' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const convert = (number) => Object.entries({
    3: "Pling",
    5: "Plang",
    7: "Plong"
  }).reduce((acc, [n, rainWord]) =>
                  acc + (isDivisibleBy(number, n) ? rainWord: ""),
            "") || String(number);

const isDivisibleBy = (n, m) => n % m === 0;
